storage/bucket.go

```diff
@@ -1608,6 +1605,11 @@ func toLifecycleFromProto(rl *storagepb.Bucket_Lifecycle) Lifecycle {
 			},
 		}
 
+		// TODO(#6205): This may not be needed for gRPC
```

Comment:
We probably need to do some deeper digging into how this works/doesn't for gRPC.

Reply:
Is there a place where metadata integration tests exist, maybe I'm not looking hard enough? Also this could wait until after this PR is merged unless there's a reason to wait to sus out gRPC first?

Reply:
Yes, this can happen later